src/bsp/platform/stm32fx_nucleo_wifi/xi_bsp_time_stm32f4_nucleo_wifi_sntp.c

```diff
@@ -121,29 +126,35 @@ static void sntp_free_response( sntp_response_t** r )
 /**
    * @brief  Create a UDP socket to the SNTP server and return its ID by
    *         setting *sock_id
-   * @param  *sntp_server is a char array with the relevant URL
-   *         sntp_port is the port to be used for SNTP communication
+   * @param  sntp_port is the port to be used for SNTP communication
    *         *sock_id will be set to the appropriate socket ID
    * @retval WiFi_Status_t: WiFi_MODULE_SUCCESS on success, see wifi_interface.h
-   *
-   * @TODO: sntp.h should have a list of SNTP servers. This function should
-   *               rotate through them, changing to the next one every time this
-   *               function is called
    */
 static WiFi_Status_t sntp_start( uint32_t sntp_port, uint8_t* sock_id )
 {
-    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
-    uint8_t sntp_protocol = 'u'; //UDP
-    if( NULL == sock_id )
+    WiFi_Status_t status                  = WiFi_MODULE_SUCCESS;
+    uint8_t sntp_protocol                 = 'u'; /* UDP */
+    static const uint8_t sntp_servers_num = sizeof( sntp_servers ) / SNTP_SERVER_HOST_LEN;
```

Comment:
something stinks here. I think
sizeof( sntp_servers ) equals to 4 x sizeof( char* ) which is 16 on a 32 byte system. 16 / 15 is 1. So this will check "only" the first two servers. At least this is my bet.

I think you could use Bob's solution from the file xi_bsp_time_cc3200_sntp.c
#define NTP_SERVER_COUNT ( sizeof( ntp_server_names ) / sizeof( char* ) )